lib/vanagon/component/dsl.rb

```diff
@@ -310,11 +310,11 @@ def build_dir(path)
 
       # This will add a source to the project and put it in the workdir alongside the other sources
       #
-      # @param url [String] url of the source
+      # @param uri [String] uri of the source
       # @param ref [String] Used for git sources, must be a git ref of some sort
       # @param sum [String] sum used to validate http and file sources
-      def add_source(url, ref: nil, sum: nil)
-        @component.sources << OpenStruct.new(:url => url, :ref => ref, :sum => sum)
+      def add_source(uri, **options)
+        @component.sources << OpenStruct.new(options)
```

Comment:
I hate that we're using OpenStruct here, because it means that there's really no documentation about what kind of object (or structure) we SHOULD be pushing into `@component.sources` here. I had to fish around for `#add_source` to figure out what is expected of each `source` we push into `@component.sources`.